libexec/daps-xmlwellformed-xinclude.xsl

```diff
@@ -33,11 +33,13 @@
   <xsl:variable name="abspath" select="s:abspath(@href)"/>
   <xsl:choose>
    <xsl:when test="s:exists(@href)">
-    <xsl:message>INFO: XIncluding <xsl:value-of select="$abspath"/></xsl:message>
+    <xsl:text>FILE:</xsl:text>
+    <xsl:value-of select="$abspath"/>
+    <xsl:text>&#10;</xsl:text>
     <xsl:apply-templates select="document($abspath, .)"/>
    </xsl:when>
    <xsl:otherwise>
-    <xsl:message>WARN: File not found "<xsl:value-of select="@href"/>"</xsl:message>
+    <xsl:message>WARN:File not found "<xsl:value-of select="@href"/>"</xsl:message>
```

Comment:
This message would be much more informative if it included the file name of the file that makes the broken include.

Reply:
That's the issue, you can't do that in pure XSLT. Maybe it's possible with an extension function.

Reply:
Ok, I've tried to get the included file name, but I'm too dump for that. :cry: 

An XPath extension function usually has this signature:

```python
def exists(context: lxml.etree._XSLTContext , nodes: list):
```

As the name implies, the function gets the context node and the arguments passed as a node list. For the context node, there is the [XPath context](https://lxml.de/extensions.html#the-xpath-context) section to get some information. To find out what's the file name of the context node, you write:

```python
# context.context_node has type lxml.etree._Element
context.context_node.base
```

However, regardless of the context node, I get always the name of the MAIN file. I would have expected to see other names... hmn, maybe my way of thinking is wrong?  :thinking:

Reply:
I don't know -- can you communicate the file name any other way? E.g. as I think you're not doing any multiprocessing, you might be able to create a custom extension function that bubbles up the current file name into a Python variable. That Python variable could then be used by your other extension function. (?)

[Hope this is a clear enough description.]

Reply:
I think, this part can be considered as resolved. :wink: